### DISCUSSION TITLE
How to find documentation

### DISCUSSION THREAD

**Discussion by josephgu41** [Field-Robotics-Japan/UnitySensors#107] [Q&A] [answered]
Hello, I am a newcomer to ROS, and I would really like to know how to achieve an effect similar to displaying GIF images in the GitHub repository. I'm wondering if there are any documents or tutorials that can guide me in creating a demo for this. Thank you! 🙏

**Comment by RyodoTanaka** [comment, marked as answer]
@josephgu41 
Thank you for your discussion.

# Current Situation
In conclusion, the GIF image on the README is a bit old version, and because of that, you could not reproduce it with the latest version.
With the latest version, we have more developed functions, such as fast, light weight, and various 3D LiDAR simulation.
Only the tutorial scene is not enough with the latest version.

# Solution
You can try sample scenes for each sensors independently.
To do that, you need to do following procedure.
1. Open the sample scene file.
2. Run the ROS-TCP-Endpoint on your ROS or ROS 2 workspace.
3. Run the scene
4. Check the results on your Rviz.

### Sample scene
To do that, open each sensors scene file such as following.
`/Assets/UnitySensorsROS/Samples/LiDAR/Livox/Demo_Mid-360_ros.unity`
You can find similar demo scene file on other directories.

### ROS / ROS 2 workspace
1. Download ROS-TCP-Endpoint on your ROS / ROS 2 workspace.
2. Checkout into correct branch. (ROS : `master`, ROS 2 : `ros2`)
3. Download dependencies using `rosdep` command.
4. Build your environment.
5. Run the launch file of endpoint (ROS : `endpoint.launch`, ROS 2 :  `endpoint.py`)

### Links of recommended reference and tips
- How to use ROS-TCP-Endpoint ? 
  The ROS-TCP-Connector is already loaded on this repository.
  So, you need to check the IP address setting and ROS-TCP-Endpoint setting.
  https://github.com/Unity-Technologies/Unity-Robotics-Hub/blob/main/tutorials/ros_unity_integration/README.md
- What is the rosdep ? 
  ROS : http://wiki.ros.org/rosdep
  ROS 2 : https://docs.ros.org/en/humble/Tutorials/Intermediate/Rosdep.html#:~:text=It%20is%20a%20command-line,install%20on%20a%20particular%20platform.
- How to build the ROS environment ? 
  http://wiki.ros.org/catkin/Tutorials
- How to build the ROS 2 environment ? 
  https://docs.ros.org/en/humble/Tutorials/Beginner-Client-Libraries/Colcon-Tutorial.html
- I could not see any information on Rviz.
  1. Check if your Global frame name on Rviz is correct or not.  
    You can find Error messages on Rviz. And inside of there, you can find what frame name is necessary.
  2. Check if `use_sim_time` parameter is `true` on ROS envoronment.
    Usually, on ROS 2 environment, you do not need to care about that. 
    However on ROS environment, you need to check the ROS time is same as simulator's time.
     To do that, `use_sim_time` parameter is used, and need to be set `true`.

**Comment by josephgu41** [comment]
Thank you very much for your response, I will try as you suggested!
